clients/src/main/java/org/apache/kafka/common/telemetry/internals/ClientTelemetryReporter.java

```diff
@@ -715,15 +716,22 @@ private Optional<Builder<?>> createPushRequest(ClientTelemetrySubscription local
             }
 
             CompressionType compressionType = ClientTelemetryUtils.preferredCompressionType(localSubscription.acceptedCompressionTypes());
-            ByteBuffer buffer = ClientTelemetryUtils.compress(payload, compressionType);
+            byte[] compressedPayload;
+            try {
+                compressedPayload = ClientTelemetryUtils.compress(payload, compressionType);
+            } catch (IOException e) {
+                log.info("Failed to compress telemetry payload for compression: {}, sending uncompressed data", compressionType);
```

Comment:
Should this be WARN-level (or maybe even ERROR) -- Do we have existing similar cases elsewhere in client for which we fall back to uncompressed?

Reply:
I think generally clients log such issues in INFO else customer opens the escalation issue where he sees logs in WARN or ERROR, that's my understanding. Maybe @philipnee @kirktrue can better address this.

I don't think there will be other scenarios where we fall back to uncompressed but client metrics is something where the compression is not defined by the user rather sent by server and automatically picked by client.

Reply:
cc: @AndrewJSchofield

Reply:
I think WARN or INFO is sufficient as it is not necessary a fatal path.  How much should user care about getting uncompressed data? If they should try to fix it maybe deserves a WARN level logging.  If this is a potential issue to the system then WARN is better. 

Also, Should we log error to DEBUG log?

Reply:
Make sense @philipnee, as user cannot fix it hence I avoided WARN or ERROR. DEBUG would just make errors on client go unnoticed hence avoided that too. I wanted to have a log which can tell that something can be fixed in clients if we see this log line.

Reply:
thanks Apoorv, i'm ok with info level logging - wdyt @mjsax? 

i mentioned logging error to debug because it seems like the caught exception is not being logged anywhere - i was wondering if we should at least somewhat log that.

Reply:
Fine with me.